### DISCUSSION TITLE
Next Mylyn Release after SimRel 2023-06

### DISCUSSION THREAD

**Discussion by ruspl-afed** [eclipse-mylyn/org.eclipse.mylyn#193] [Polls]
Let's have a survey how do we see Mylyn after 2023-06, 
since there are a lot of pending changes that includes "breaking", like repackaging/renaming. 
On the other hand, our resources are limited, and we are unlikely to be able to do a noticeable amount of work over the next three months.

**Comment by jonahgraham** [comment]
IMHO you should do a 4.0.0 release soon, and set the expectation that you will do a 5.0.0 release fairly soon after that with an announced expectation that Mylyn will stay on 5.x as long as possible (i.e many years). I hope this will help spur people to get involved in the 4.x release time frame for two reasons:

1. Extenders/API consumers being more vocal about API they care about.
2. Significantly aides contributors to Mylyn by providing a break with the past. Maintaining API compatibility often requires significant work and slows down velocity on new work

**Reply by gnl42** [reply]
What do you see being in 4.0 that wouldn't be in 5.0?

**Reply by jonahgraham** [reply]
My comment is related to the tight binding between project versions and semantic versioning. So 5.0 would allow API changes that simplifies development.

**Comment by ruspl-afed** [comment]
Thanks to everyone who left their vote, let's extend this poll till 12th of June and then start another release cycle.
